config/tectonic-node-controller/config.go

```diff
@@ -32,6 +32,9 @@ type ControllerConfig struct {
 
 	// User customizations, list of node configs in the cluster to collect and apply as part of the final config
 	AdditionalConfigs []string `json:"additionalConfigs"`
+
+	// Pool update percentages allowing individual roles to have a specific percentage of nodes that can be upgrading at a given time ie. worker=10%, if 0% only a single node will be updated at a given time
+	NodePoolUpdatePercent map[string]int
```

Comment:
json tag?

Reply:
@aaronlevy OK to include this knob?

Reply:
Yeah, I think it's reasonable - however, I think we need to allow absolute as well as percentages. Reason for this are things like etcd (only 1 node at a time). Maybe we do this similar to deployments `maxUnavailable` (which can be percent or absolute number). See: https://jenkins-kube-lifecycle.prod.coreos.systems/job/tectonic-operators/job/master/

Thoughts?

Reply:
Sounds reasonable, then we need to change this to a string, the format will be either integer (e.g. `1`) or percentage (e.g. `10%`). Maybe also rename it to `NodePoolUpdateLimit` ? @thorfour

Reply:
That sounds reasonable